app/services/return-requirements/generate-return-version-requirements.service.js

```diff
@@ -0,0 +1,133 @@
+      legacyId,
+      reabstraction: requirement.agreementsExceptions.includes('transfer-re-abstraction-scheme'),
+      reportingFrequency: requirement.frequencyReported,
+      returnsFrequency: 'year',
+      returnRequirementPoints: _generateReturnRequirementPoints(licencePoints, externalId, requirement.points),
+      returnRequirementPurposes: await _generateReturnRequirementPurposes(licenceId, requirement.purposes),
+      siteDescription: requirement.siteDescription,
+      summer: requirement.returnsCycle === 'summer',
+      twoPartTariff: requirement.agreementsExceptions.includes('two-part-tariff')
+    }
+
+    legacyId++
+
+    returnRequirements.push(returnRequirement)
+  }
+
+  return returnRequirements
+}
+
+async function _fetchNaldRegionId (licenceId) {
+  const { naldRegionId } = await LicenceModel.query()
+    .findById(licenceId)
+    .select('region.naldRegionId')
+    .innerJoinRelated('region')
+
+  return naldRegionId
+}
+
+function _generateReturnRequirementPoints (licencePoints, requirementExternalId, requirementPoints) {
+  const returnRequirementPoints = []
+
+  requirementPoints.forEach((requirementPoint) => {
+    const point = licencePoints.find((licencePoint) => {
+      return licencePoint.ID === requirementPoint
+    })
+
+    const returnRequirementPoint = {
+      description: point.LOCAL_NAME,
+      externalId: `${requirementExternalId}:${point.ID}`,
+      naldPointId: point.ID,
+      ngr1: point.NGR1_SHEET !== 'null' ? `${point.NGR1_SHEET} ${point.NGR1_EAST} ${point.NGR1_NORTH}` : null,
+      ngr2: point.NGR2_SHEET !== 'null' ? `${point.NGR2_SHEET} ${point.NGR2_EAST} ${point.NGR2_NORTH}` : null,
+      ngr3: point.NGR3_SHEET !== 'null' ? `${point.NGR3_SHEET} ${point.NGR3_EAST} ${point.NGR3_NORTH}` : null,
+      ngr4: point.NGR4_SHEET !== 'null' ? `${point.NGR4_SHEET} ${point.NGR4_EAST} ${point.NGR4_NORTH}` : null
+    }
+
+    returnRequirementPoints.push(returnRequirementPoint)
+  })
+
+  return returnRequirementPoints
+}
+
+async function _generateReturnRequirementPurposes (licenceId, purposeIds) {
+  const returnRequirementPurposes = []
+
+  for (const purposeId of purposeIds) {
+    const { primaryPurposeId, secondaryPurposeId } = await LicenceVersionModel.query()
+      .select('primaryPurposeId', 'secondaryPurposeId')
+      .innerJoinRelated('licenceVersionPurposes')
+      .where('licenceId', licenceId)
+      .andWhere('status', 'current')
+      .andWhere('purposeId', purposeId)
+      .first()
+
+    const returnRequirementPurpose = {
+      primaryPurposeId,
+      purposeId,
+      secondaryPurposeId
+    }
+
+    returnRequirementPurposes.push(returnRequirementPurpose)
+  }
+
+  return returnRequirementPurposes
+}
+
+async function _getNextLegacyId () {
+  const { lastLegacyId } = await ReturnRequirementModel.query()
+    .max('legacyId as lastLegacyId')
+    .first()
+
+  if (lastLegacyId) {
+    return lastLegacyId + 1
+  }
+
+  return 1
+}
```

Comment:
I'm afraid it's more complicated than this.

`legacy_id` Is an incremental ID by region. In order to figure it out you need to

- Select all return requirements for the same region
- Select the MAX(legacy ID) from the results
- increment the result by 1

Selecting by region can be achieved with a

```suggestion
async function _nextLegacyId (naldRegionId) {
  const { lastLegacyId } = await ReturnRequirementModel.query()
    .max('legacyId as lastLegacyId')
    .where('externalId, 'like', `${naldRegionId}:%`)
    .first()

  if (lastLegacyId) {
    return lastLegacyId + 1
  }

  return 1
}
```

Also, it's been our convention (where we spot it!) _not_ to prefix functions with `_get` or `_set` hence the tweak to the method name.